Wikipedia/Code/WMFFeedContentSource.m

```diff
@@ -232,10 +232,28 @@ - (void)saveGroupForPictureOfTheDay:(WMFFeedImage *)image date:(NSDate *)date in
 
     WMFContentGroup *group = [self pictureOfTheDayForDate:date inManagedObjectContext:moc];
 
-    if (group == nil) {
-        group = [moc createGroupOfKind:WMFContentGroupKindPictureOfTheDay forDate:date withSiteURL:self.siteURL associatedContent:nil];
+    NSString *primaryLanguage = self.userDataStore.languageLinkController.appLanguage.languageCode;
+    NSString *feedLanguage = [self.siteURL wmf_languageCode];
+    BOOL isPrimaryLanguageFeed = [feedLanguage isEqualToString:primaryLanguage];
+
+    // T356255 - the POTD is the same across all feeds (i.e. same image, different localized description).
+    // To prevent the same image from showing on each language feed, we save the POTD for the primary language only
+    // (and clean up any secondary ones - secondary ones may exist when the user has changed the app primary language).
+
+    if (isPrimaryLanguageFeed) {
+        if (group == nil) {
+            DDLogDebug(@"Creating POTD content group for primary feed language [%@]", feedLanguage);
+            group = [moc createGroupOfKind:WMFContentGroupKindPictureOfTheDay forDate:date withSiteURL:self.siteURL associatedContent:nil];
+        } else {
+            DDLogDebug(@"Updating POTD content group for primary feed language [%@]", feedLanguage);
+        }
+        group.contentPreview = image;
+    } else {
+        if (group != nil) {
+            DDLogDebug(@"Removing POTD content group for OLD secondary feed language [%@].", feedLanguage);
+            [moc removeContentGroup:group];
+        }
```

Comment:
This is the main logic change (1 of 2). 

Here's what I've done:

- The original logic is still the same (it's just moved inside the `if (isPrimaryLanguageFeed) { }` statement)
- The `else` part is new and prevents duplicates by removing any POTD records found against non-primary languages.
- The logging statements are new and were really just to help me troubleshoot (please let me know if you'd like me to remove them).